libyul/optimiser/CommonSubexpressionEliminator.h

```diff
@@ -52,6 +57,9 @@ class CommonSubexpressionEliminator: public DataFlowAnalyzer
 protected:
 	using ASTModifier::visit;
 	void visit(Expression& _e) override;
+
+private:
+	std::stack<std::set<YulString>> m_returnVariables;
```

Comment:
Why a stack? Wouldn't saving and restoring in the function definition visit be enough?

Reply:
True!